llvm/lib/CodeGen/CodeGenPrepare.cpp

```diff
@@ -1974,6 +1975,161 @@ static bool foldFCmpToFPClassTest(CmpInst *Cmp, const TargetLowering &TLI,
   return true;
 }
 
+static bool isRemOfLoopIncrementWithLIV(Value *Rem, const LoopInfo *LI,
+                                        Value *&RemAmtOut,
+                                        std::optional<bool> &AddOrSubOut,
+                                        Value *&AddOrSubOffsetOut,
+                                        PHINode *&LoopIncrPNOut) {
+  Value *Incr, *RemAmt;
+  if (!isa<Instruction>(Rem))
+    return false;
```

Comment:
Please add a negative test with urem constexpr.

Reply:
Is that necessary? We are trying to remove ConstantExpr and this doesn't seem so complex that it might fail.